resolver-dns/src/main/java/io/netty/resolver/dns/NameServerCache.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver.dns;
+
+import io.netty.util.internal.UnstableApi;
+
+/**
+ * Provides an opportunity to override which {@link DnsServerAddressStream} is used to resolve a specific hostname.
+ * <p>
+ * For example this can be used to represent <a href="https://linux.die.net/man/5/resolver">/etc/resolv.conf</a> and
+ * <a href="https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man5/resolver.5.html">
+ * /etc/resolver</a>.
+ */
+@UnstableApi
+public interface NameServerCache {
```

Comment:
`DnsCache` is designed to cache "resolved addresses" where as this is designed to cache DNS resolver addresses obtained from the [resolver configuration files](https://linux.die.net/man/5/resolver). So the concepts are a bit different and the interface for `DnsCache` supports stuff which are unnecessary for our purposes (e.g. `DnsRecord[] additionals`).

Here are some options:
1. The name I choose (`NameServerCache`) is bad and could be clarified. Here are some options:
`LocalHostNameServerConfig`
`ResolverConfigCache`
`ResolverConfigurationFileCache`
2. Extract out some common interface from `DnsCacheEntry` (maybe just the `get(..)` method) into some base interface (e.g. `ReadOnlyDnsCache`). I'm not sure if this is worth the trouble.